students/charlie_england/lesson05/assignment/src/database.py

```diff
@@ -0,0 +1,99 @@
+import csv
+import os.path
+from pymongo import MongoClient
+
+mongo = MongoClient("localhost:27017")
+
+db = mongo["hp_norton"]
+db.drop_collection("product")
+db.drop_collection("customers")
+db.drop_collection("rental")
+
+def import_data(data_dir, *files):
+    """
+        given a directory and files
+    """
+    added = [0, 0, 0]
+    errors = [0, 0, 0]
+    reference = 0
+    for filepath in files:
+        collection_name = filepath.split(".")[0]
+        with open(os.path.join(data_dir, filepath)) as file:
+            reader = csv.reader(file, delimiter=",")
+            header = False
+
+            for row in reader:
+                try:
+                    if not header:
+                        header = [h.strip("\ufeff") for h in row]
+                    else:
+                        data = {header[i]:v for i, v in enumerate(row)}
+                        cursor = db[collection_name]
+                        cursor.insert_one(data)
+                        added[reference] +=1
+                except Exception:
+                    errors[reference] += 1
+            reference += 1
+    return tuple(added), tuple(errors)
+
+def show_available_products():
+    """
+        Returns a Python dictionary of products listed as available with the following fields:
+        product_id.
+        description.
+        product_type.
+        quantity_available.
+        For example: {‘prd001’:{‘description’:‘60-inch TV stand’,
+        ’product_type’:’livingroom’,’quantity_available’:‘3’},
+        ’prd002’:{‘description’:’L-shaped sofa’,
+        ’product_type’:’livingroom’,’quantity_available’:‘1’}}
+    """
+    col = db["product"]
+    query = {"quantity_available":{"$gt":"0"}}
+    final_dict = {}
+    available_products = iter(col.find(query))
+    for product in available_products:
+        prod_id = product["ï»¿product_id"]
```

Comment:
This looks like you have some sort of encoding issue, you should be able to fix this if you read that string as str(string_variable, "utf-8")